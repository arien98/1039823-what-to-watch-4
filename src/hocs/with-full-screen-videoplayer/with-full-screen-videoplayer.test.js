import React from "react";
import renderer from "react-test-renderer";
import withVideo from "../with-video/with-video.jsx";

const MockComponent = () => {
  return (
    <div>
    </div>
  );
};

const MockComponentWrapped = withVideo(MockComponent);

describe(`with full screen videoplayer snapshot`, () => {
  test(`it renders correctly`, () => {
    const filmData = {
      title: `Aviator`,
      smallPoster: `../img/aviator.jpg`,
      genre: `Drama`,
      releaseDate: 2010,
      preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    };

    const tree = renderer
    .create(
        <MockComponentWrapped
          filmData={filmData}
          onFilmCardClick={() => {}}
        />, {
          createNodeMock: (element) => {
            if (element.type === `video`) {
              return {
                play: () => {},
                src: `ddd`,
                poster: `fff`,
              };
            }
            return null;
          }
        }
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
