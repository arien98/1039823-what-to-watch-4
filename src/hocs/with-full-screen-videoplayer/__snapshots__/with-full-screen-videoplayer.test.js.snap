// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with full screen videoplayer snapshot it renders correctly 1`] = `<div />`;
